static/js/message_events.js

```diff
@@ -393,11 +393,11 @@ export function update_messages(events) {
     // propagated edits to be updated (since the topic edits can have
     // changed the correct grouping of messages).
     if (topic_edited || stream_changed) {
-        home_msg_list.update_muting_and_rerender();
+        home_msg_list.update_and_rerender_for_topic_muting();
         // However, we don't need to rerender message_list.narrowed if
         // we just changed the narrow earlier in this function.
         if (!changed_narrow && current_msg_list === message_list.narrowed) {
-            message_list.narrowed.update_muting_and_rerender();
+            message_list.narrowed.update_and_rerender_for_topic_muting();
```

Comment:
Can you pull this commit out in a separate PR (or just move it to be the first commit on the branch)?  We can merge it right now, since it's a helpful change in any case.

(Though I think I'd prefer the name `update_topic_muting_and_rerender`).

Reply:
Shifted this commit to the beginning of the PR, and also pulled out a couple other refactoring changes from the main commits 🙂